src/routers/comment-router.js

```diff
@@ -0,0 +1,95 @@
+const express = require("express");
+const router = express.Router();
+const { check, validationResult } = require("express-validator");
+const Comment = require("../models/Comment");
+const comment_controller = require("../controllers/comment-controller");
+
+//Comment Routes
+
+/*
+Routes GET requests at root 
+returns a promise of a JSON object containing a list of comments
+*/
+router.get("/", function(req, res) {
+  const { userID, username, text, date } = req.body;
+
+  Comment.find({
+    $or: [{ userID: userID }, { text: text }, { date: date }]
+  })
+    .lean()
```

Comment:
What is the `.lean()` supposed to do?

Reply:
According to the docs
> This makes queries faster and less memory intensive, but the result documents are plain old JavaScript objects (POJOs), not Mongoose documents. In this tutorial, you'll learn more about the tradeoffs of using lean().

The downside of enabling lean is that lean docs don't have:
Change tracking
Casting and validation
Getters and setters
Virtuals
save()

Reply:
Got it.